### PR TITLE
[ci] move rllib small-gpu tests to civ2

### PR DESCRIPTION
**Pull request by can-anyscale**
- rllib uses 'gpu' and 'multi-gpu' tag which confuses bazel query tag since it takes the input as a regex. Updated ray_ci to strictly query for an exact match of tag
- we cannot parallelize this job because the machine has only one core

Test:
- CI

### CODE REVIEW

File: .buildkite/rllib.rayci.yml [hunk 1]
```diff
@@ -77,6 +77,20 @@ steps:
     depends_on: rllibbuild
     job_env: forge
 
+  - label: ":brain: rllib: gpu tests"
+    tags: rllib
```
Comment (aslonnie) [inline .buildkite/rllib.rayci.yml:81]: add gpu tag?

File: ci/ray_ci/tester.py [hunk 2]
```diff
@@ -160,6 +160,17 @@ def _get_container(
...
 
+def _get_tag_matcher(tag: str) -> str:
+    """
+    Return a regular expression that matches the given bazel tag. This is required for
+    an exact tag match because bazel query uses regex to match tags.
+
+    The word boundary is escaped twice because it is used in a python string and then
+    used again as a string in bazel query.
+    """
+    return f"\\\\b{tag}\\\\b"
```
Comment (aslonnie) [inline ci/ray_ci/tester.py:171]: could you add a unit test for this function? just to illustrate how this works?

File: ci/ray_ci/test_tester.py [hunk 3]
```diff
@@ -11,9 +12,23 @@
...
     _get_test_targets,
     _get_flaky_test_targets,
+    _get_tag_matcher,
 )
 
 
+def test_get_tag_matcher() -> None:
+    assert re.match(
+        # simulate shell character escaping
+        bytes(_get_tag_matcher("tag"), "utf-8").decode("unicode_escape"),
```
Comment (aslonnie) [inline ci/ray_ci/test_tester.py:22]: why is this using `unicode_escape`?

**Review by aslonnie — APPROVED** [review]

File: ci/ray_ci/test_tester.py (hunk 3 above)
Comment (can-anyscale) [reply ci/ray_ci/test_tester.py:22]: It evaluates the escape character in a string, for example, the string 'a\\nb' will be evaluated to 'a\nb', which is what happen to a string in a shell command

**Pull request merged**